### PR TITLE
Workaround for unknown ALERT{} label length

### PR DESCRIPTION
**Pull request by erdii**
Heyo :wave: 

I'm currently working on incorporating jiralert in our alerting-pipeline and have repeatedly run into a problem: [JIRA has a length limit of 255 chars for a single issue label](https://docs.atlassian.com/software/jira/docs/api/7.1.8/com/atlassian/jira/bc/issue/label/LabelService.html#:~:text=The%20label%20will%20also%20be,max%20length%20of%20255%20characters.)

Since the `ALERT{...}` label is not intended for human consumption, I would like to propose using a computed hash value instead:

`ALERT{` +  `sha512sum(groupLabels)` + `}`

this way, it is ensured, that jiralert will never pass an invalid label because the length is fixed.

---

What do you think? Do you think this breaking change is acceptable?

### CODE REVIEW

**Comment by erdii** [comment]
I've prepared a commit that by default disables the hashing behavior and add a flag `--hash-jira-label` that toggles this behavior on: https://github.com/erdii/jiralert/commit/709239d7be514f02cd3a958c4ec9dc20ec12fe7d

**Comment by bwplotka** [comment]
Nice! I will look at it, feel free to DM me offline as well 🤗

File: pkg/notify/notify.go [hunk 1]
```diff
@@ -225,16 +224,17 @@ func deepCopyWithTemplate(value interface{}, tmpl *template.Template, data inter
...
+// if the combined length of all groupLabel key-value pairs would be
+// longer than 255 chars
 func toGroupTicketLabel(groupLabels alertmanager.KV) string {
-	buf := bytes.NewBufferString("ALERT{")
+	hash := sha512.New()
 	for _, p := range groupLabels.SortedPairs() {
-		buf.WriteString(p.Name)
-		buf.WriteString(fmt.Sprintf("=%q,", p.Value))
+		kvString := fmt.Sprintf("%s=%q,", p.Name, p.Value)
+		hash.Write([]byte(kvString)) // nolint:errcheck // hash.Write can never return an error
```
Comment (bwplotka) [inline pkg/notify/notify.go:235]: Well done on nolint and comment 🤗

**Review by bwplotka — APPROVED** [review]
Thanks for this!

That will work, just wonder if that statement is true:

> Since the ALERT{...} label is not intended for human consumption, I would like to propose using a computed hash value instead:

🤗 Such label will scare people... but maybe there is no better choice. We could put something like this in Ticket description e.g, and do filtering on our side. I think I am find with this for now (: 

LGTM!

File: pkg/notify/notify.go (hunk 1 above)
Comment (bwplotka) [reply pkg/notify/notify.go:235]: Actually, there might better way! (: 
```suggestion
		_ = hash.Write([]byte(kvString)) 
```

**Comment by erdii** [comment]
will hide new behavior behind flag and in the long term deprecate the old behavior and the flag

**Comment by erdii** [comment]
I added the flag `-hash-jira-label` added a doc blurb to README.md and changed `ALERT{...}` to `JIRALERT{...}` to clarify the origin of this label on a jira issue.

This is ready for another review @bwplotka :)

File: README.md [hunk 2]
```diff
@@ -52,6 +52,9 @@ $ curl -H "Content-type: application/json" -X POST \
 
 ## Configuration
 
+Attention: We have introduced a new behavior in [#79](https://github.com/prometheus-community/jiralert/pull/79) for building the string that is used as an issue label in jira so jiralert can find an issue it opened. The new behavior ensures that at all times the string is a legal issue label in the jira api. **This change is opt-in** to avoid breaking users upgrading from older releases, but we will remove the old behavior + flag in a future release.
```
Comment (bwplotka) [inline README.md:55]: That should probably go to CHANGELOG.md instead 🤗

**Review by bwplotka — APPROVED** [review]
Thanks! Let's put one thing to changelog instead and then LGTM!

File: README.md (hunk 2 above)
Comment (erdii) [reply README.md]: - #83 adds a changelog and is now a prerquisite for this pr
- this pr adds a new entry in the changelog

**Review by bwplotka — APPROVED** [review]
LGTM, just a rebase needed 🤗

**Comment by erdii** [comment]
:tada: thank you for your reviews and guidance @bwplotka <3

**Pull request merged**